titus-server-master/src/main/java/com/netflix/titus/master/mesos/kubeapiserver/direct/DefaultDirectKubeApiServerIntegrator.java

```diff
@@ -70,6 +80,13 @@ public DefaultDirectKubeApiServerIntegrator(DirectKubeConfiguration configuratio
         this.apiClient = apiClient;
         this.coreV1Api = new CoreV1Api(apiClient);
         this.taskToPodConverter = taskToPodConverter;
+
+        ThreadFactory threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("kube-apiclient-%s")
+                .setDaemon(true)
+                .build();
+        this.apiClientExecutor = Executors.newFixedThreadPool(configuration.getApiClientThreadPoolSize(), threadFactory);
```

Comment:
Use one of the `ExecutorExt.instrumented...ThreadPool()` options to get metrics here